itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenarios.java

```diff
@@ -2069,6 +2098,194 @@ public void testTruncateWithCM() throws IOException {
+
+    // INSERT EVENT to partitioned table with dynamic ADD_PARTITION
+    run("INSERT INTO TABLE " + dbName + ".ptned PARTITION(b=1) values('" + ptn_data_1[0] + "')");
+    replDump = dumpDbFromLastDump(dbName, replDump);
+    incrementalDumpList.add(replDump);
+
+    // ADD_PARTITION EVENT to partitioned table
+    run("ALTER TABLE " + dbName + ".ptned ADD PARTITION (b=2)");
+    replDump = dumpDbFromLastDump(dbName, replDump);
+    incrementalDumpList.add(replDump);
+
+    // INSERT EVENT to partitioned table on existing partition
+    run("INSERT INTO TABLE " + dbName + ".ptned PARTITION(b=2) values('" + ptn_data_2[0] + "')");
+    replDump = dumpDbFromLastDump(dbName, replDump);
+    incrementalDumpList.add(replDump);
+
+    // TRUNCATE_PARTITION EVENT on partitioned table
+    run("TRUNCATE TABLE " + dbName + ".ptned PARTITION (b=1)");
+    replDump = dumpDbFromLastDump(dbName, replDump);
+    incrementalDumpList.add(replDump);
+
+    // TRUNCATE_TABLE EVENT on unpartitioned table
+    run("TRUNCATE TABLE " + dbName + ".unptned");
+    replDump = dumpDbFromLastDump(dbName, replDump);
+    incrementalDumpList.add(replDump);
+
+    // CREATE_TABLE EVENT with multiple partitions
+    run("CREATE TABLE " + dbName + ".unptned_tmp AS SELECT * FROM " + dbName + ".ptned");
+    replDump = dumpDbFromLastDump(dbName, replDump);
+    incrementalDumpList.add(replDump);
+
+    // Replicate all the events happened so far
+    Tuple incrDump = incrementalLoadAndVerify(dbName, bootstrapDump.lastReplId, replDbName);
+
+    verifyRun("SELECT a from " + replDbName + ".unptned ORDER BY a", empty);
+    verifyRun("SELECT a from " + replDbName + ".ptned where (b=1) ORDER BY a", empty);
+    verifyRun("SELECT a from " + replDbName + ".ptned where (b=2) ORDER BY a", ptn_data_2);
+    verifyRun("SELECT a from " + replDbName + ".unptned_tmp where (b=1) ORDER BY a", empty);
+    verifyRun("SELECT a from " + replDbName + ".unptned_tmp where (b=2) ORDER BY a", ptn_data_2);
+
+    // Load each incremental dump from the list. Each dump have only one operation.
+    for (Tuple currDump : incrementalDumpList) {
+      // Load the incremental dump and ensure it does nothing and lastReplID remains same
+      loadAndVerify(replDbName, currDump.dumpLocation, incrDump.lastReplId);
+
+      // Verify if the data are intact even after applying an applied event once again on existing objects
+      verifyRun("SELECT a from " + replDbName + ".unptned ORDER BY a", empty);
+      verifyRun("SELECT a from " + replDbName + ".ptned where (b=1) ORDER BY a", empty);
+      verifyRun("SELECT a from " + replDbName + ".ptned where (b=2) ORDER BY a", ptn_data_2);
+      verifyRun("SELECT a from " + replDbName + ".unptned_tmp where (b=1) ORDER BY a", empty);
+      verifyRun("SELECT a from " + replDbName + ".unptned_tmp where (b=2) ORDER BY a", ptn_data_2);
+    }
+  }
+
+  @Test
+  public void testIncrementalRepeatEventOnMissingObject() throws IOException {
```

Comment:
Sorry, this test I don't think appropriate to separate for partitioned and unpartitioned. Just care about the sequence of operations. Also, I need multiple tables to verify the change.

About moving to TestReplicationScenariosAcrossInstances, I think, the idea is deferred due to cmroot file system issues and we don't have any plan to fix that now. Please create a JIRA and will plan it in future.